modules/account-lib/src/coin/sol/walletInitializationBuilder.ts

```diff
@@ -1,37 +1,65 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { Transaction } from './transaction';
 import { TransactionBuilder } from './transactionBuilder';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { WalletInit } from './iface';
+import { InstructionBuilderTypes } from './constants';
 
 export class WalletInitializationBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
-
   protected get transactionType(): TransactionType {
     return TransactionType.WalletInitialization;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
-  }
+  /**
+   *  Creates a nonce account for an address
+   *
+   * @param {string} fromAddress - the address whos gonna own the nonce
+   * @param {string} nonceAddress - the nonce address to be created
+   * @param {string} authAddress - the address whos gonna own the nonce
```

Comment:
doc for `fromAddress` and `authAddress` are the same.

consider renaming `fromAddress` to `feePayer`

Reply:
The question here is, do we want to be able to set a fromAddress(the one who will send the lamports to cover the nonce account rent) that can be different than authAddress (the owner of the nonce account)?

Reply:
Removed fromAddress